### PR TITLE
PHPBench performace testing

### PR DESCRIPTION
**Pull request by mrceperka**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -40,3 +40,5 @@ mutations: ## Check code for mutants
 		--coverage=../var/coverage \
 		--skip-initial-tests \
 		$(ARGS)
+bench:
```
Comment (mabar) [inline Makefile]: `bench: ## Performance benchmark`
- type `make` it lists all the by comment documented commands :)

File: Makefile [hunk 2]
```diff
@@ -40,3 +40,5 @@ mutations: ## Check code for mutants
...
+	vendor/bin/phpbench run tests/Benchmark --report=default
```
Comment (mabar) [inline Makefile]: Please sync it with how other tools are used currently - `$(PRE_PHP)` before command to disable xdebug +  variable with path to config file + `$(ARGS)` to be able append any arguments (e.g. for CI needs)

File: composer.json [hunk 3]
```diff
@@ -29,6 +29,7 @@
 		"nette/caching": "^3.0",
 		"nette/di": "^3.0.0",
 		"orisai/coding-standard": "^2.0.0",
+		"phpbench/phpbench": "^1.0",
```
Comment (mabar) [inline composer.json]: `^1.0.0`
- it's not that case, but Composer has some odd cases with `x.y` syntax

File: phpbench.json [hunk 4]
```diff
@@ -0,0 +1,3 @@
+{
+	"runner.bootstrap": "vendor/autoload.php"
+}
```
Comment (mabar) [inline phpbench.json]: Move config file to tools directory

File: tests/Benchmark/ArrayOfBench.php [hunk 5]
```diff
@@ -0,0 +1,59 @@
+<?php declare(strict_types = 1);
+
+namespace Orisai\Tests\Benchmark;
```
Comment (mabar) [inline tests/Benchmark/ArrayOfBench.php]: Should be `Tests\Orisai\ObjectMapper\Benchmark`
PHPStorm should be able autocomplete namespaces if you enable composer.json sync feature
Also `make cs` would tell you

File: tests/Benchmark/ArrayOfBench.php [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+
+use Generator;
+use Orisai\ObjectMapper\Exception\InvalidData;
+use Tests\Orisai\ObjectMapper\Doubles\ArrayOfIntVO;
+use Tests\Orisai\ObjectMapper\Doubles\ArrayOfStringVO;
+use Tests\Orisai\ObjectMapper\Toolkit\ProcessingTestCase;
+use function array_fill;
+
+final class ArrayOfBench extends ProcessingTestCase
```
Comment (mabar) [inline tests/Benchmark/ArrayOfBench.php:12]: Shouldn't extends phpunit test case, but I see the reason. Will solve that myself, some setup for apps without DIC gonna be helpful anyway 
#8

Comment (mrceperka) [reply tests/Benchmark/ArrayOfBench.php:12]: Rebased and rest of the issues resolved :crossed_fingers:

**Comment by mabar** [comment]
Epic 👌

**Pull request merged**